apps/buildog/src/app/(blog-pages)/blog/page.tsx

```diff
@@ -52,48 +50,64 @@ export default function Page() {
+    };
+  };
+
   return (
     <div className="space-y-4">
-      <Input />
-      <div className="grid grid-cols-4 gap-4">
-        {blogs ? (
-          Object.entries(blogs).map(([key, blog]) => (
-            <Card key={key}>
-              <CardHeader>
-                <CardTitle>{blog.filename}</CardTitle>
-                <Button
-                  variant="destructive"
-                  className="w-10 ml-auto"
-                  onClick={() => setOpen(!open)}
-                  size="sm"
-                >
-                  <Cross1Icon />
-                </Button>
-                {/* <CardDescription>{blog}</CardDescription> */}
-              </CardHeader>
-              <Dialog onOpenChange={setOpen} open={open}>
-                <DialogContent className="sm:max-w-[425px]">
-                  <DialogHeader>
-                    <DialogTitle>Delete Blog</DialogTitle>
-
-                    <DialogDescription>Are you sure for delete this blog?</DialogDescription>
-                  </DialogHeader>
-                  <DialogFooter>
-                    <Button variant="ghost" onClick={() => setOpen(!open)}>
-                      No
-                    </Button>
-                    <Button variant="destructive" onClick={() => deleteBlog(blog.filename)}>
-                      Yes
+      <DataTable
+        columns={[
+          {
+            id: "fileName",
+            accessorKey: "fileName",
+            header: "Blog Name",
+          },
+          {
+            accessorKey: "blogId",
+            header: "Blog Id",
+          },
+          {
+            accessorKey: "tags",
+            header: "Tags",
+          },
+          {
+            accessorKey: "actions",
+            header: "Actions",
+            id: "actions",
+            cell: ({ row }) => {
+              const fileName = row.original.fileName;
+              return (
+                <DropdownMenu>
+                  <DropdownMenuTrigger asChild>
+                    <Button variant="ghost" className="h-8 w-8 p-0">
+                      ...
                     </Button>
-                  </DialogFooter>
-                </DialogContent>
-              </Dialog>
-            </Card>
-          ))
-        ) : (
-          <></>
-        )}
-      </div>
+                  </DropdownMenuTrigger>
+                  <DropdownMenuContent align="end">
+                    <DropdownMenuLabel>Actions</DropdownMenuLabel>
+                    <DropdownMenuItem>Edit</DropdownMenuItem>
+                    <DropdownMenuSeparator />
+                    <DropdownMenuItem
+                      onClick={() => {
+                        deleteBlog(fileName);
+                      }}
+                    >
+                      Delete
+                    </DropdownMenuItem>
+                  </DropdownMenuContent>
+                </DropdownMenu>
+              );
+            },
+          },
+        ]}
+        data={Object.values(blogs || {}).map(createData)}
```

Comment:
This idea works fine but, readability is less. Separating map's callback function is not good. You can directly use array function inside the map. 

```TypeScript
Object.values(blogs || {}).map((blog)=> {
       blogId: "",
       fileName: blog.filename,
       tags: "",
     })
```

or use;

```TypeScript

const tableData = () => {
    return Object.values(blogs || {}).map((blog)=> {
       blogId: "",
       fileName: blog.filename,
       tags: "",
     })
}

<DataTable
     ...
     data={tableData()}
     ...
/>

```